zio-sql/sql/shop.sql

```diff
@@ -0,0 +1,199 @@
+use shop
+
+drop table users
+drop table orders
+drop table products
+drop table product_prices
+drop table order_details
+
+create table users
+(
+    usr_id uniqueidentifier primary key default newid() not null,
+    first_name nvarchar(255) not null,
+    last_name nvarchar(255) not null,
+    dob date not null,
+)
+
+create table orders
+(
+    order_id uniqueidentifier primary key default newid() not null,
+    usr_id uniqueidentifier not null,
+    order_date date not null,
+)
+
+create table products
+(
+    product_id uniqueidentifier primary key default newid() not null,
+    product_name nvarchar(255),
+    description nvarchar(max) not null,
+    image_url nvarchar(max),
+)
+
+create table product_prices
+(
+    product_id uniqueidentifier not null,
+    effective date not null,
+    price money not null
+)
+
+create table order_details
+(
+    order_id uniqueidentifier not null,
+    product_id uniqueidentifier not null,
+    quantity integer not null,
+    unit_price money not null
+)
+
+
+insert users
+    (usr_id, first_name, last_name, dob)
+values
+    ('60b01fc9-c902-4468-8d49-3c0f989def37', 'Ronald', 'Russell', '1983-01-05'),
```

Comment:
Just checking this is all fake data and not previously owned test data IP, right? 😉

Reply:
Wrote it all myself or randomly generated e.g. order details started life like this before I grouped them

```sql
insert into order_details
select top 5
    order_id,
    product_id,
    1 as quantity,
    price
from orders o cross apply(
select top 1 *
    from product_prices pp
    where pp.effective <= o.order_date
    order by effective desc, newid() ) a
order by newid()
```